crates/net/eth-wire/src/types/broadcast.rs

```diff
@@ -104,16 +104,46 @@ pub struct SharedTransactions(
 #[derive_arbitrary(rlp)]
 #[derive(Clone, Debug, PartialEq, Eq, RlpEncodableWrapper, RlpDecodableWrapper, Default)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-pub struct NewPooledTransactionHashes(
+pub struct NewPooledTransactionHashes66(
     /// Transaction hashes for new transactions that have appeared on the network.
     /// Clients should request the transactions with the given hashes using a
     /// [`GetPooledTransactions`](crate::GetPooledTransactions) message.
     pub Vec<H256>,
 );
 
-impl From<Vec<H256>> for NewPooledTransactionHashes {
+impl From<Vec<H256>> for NewPooledTransactionHashes66 {
     fn from(v: Vec<H256>) -> Self {
-        NewPooledTransactionHashes(v)
+        NewPooledTransactionHashes66(v)
+    }
+}
+
+/// Same as [`NewPooledTransactionHashes66`] but extends that that beside the transaction hashes,
+/// the node sends the transaction types and their sizes (as defined in EIP-2718) as well.
+#[derive_arbitrary(rlp)]
+#[derive(Clone, Debug, PartialEq, Eq, RlpEncodable, RlpDecodable, Default)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct NewPooledTransactionHashes68 {
+    /// Transaction types for new transactions that have appeared on the network.
+    pub types: Vec<u8>,
+    /// Transaction sizes for new transactions that have appeared on the network.
+    pub sizes: Vec<usize>,
+    /// Transaction hashes for new transactions that have appeared on the network.
+    pub hashes: Vec<H256>,
+}
+
+impl TryFrom<(Option<Vec<u8>>, Option<Vec<usize>>, Vec<H256>)> for NewPooledTransactionHashes68 {
```

Comment:
not a fan of this conversion.

why would we need this anyway?
I'd like to remove this.

Reply:
Please give me an advice. T_T

* It's used for converting from `NewPooledTransactionHashes` into `NewPooledTransactionHashes68`.
* `NewPooledTransactionHashes` is defined in `network` so we could not define `TryFrom<NewPooledTransactionHashes> for NewPooledTransactionHashes68`.
* The only alternative idea I have is that `pub fn new(...) -> Result<Self, Error>`.

HDYT?

Reply:
do we need this in this PR?

I still don't understand why we would need this.

in any case, a TryFrom a tuple with 3 items is not great.